### ISSUE TITLE
Improve CI build times, esp. on Ubuntu

### ISSUE THREAD

**Issue by repi** [EmbarkStudios/rust-gpu#107]
These are our current CI build times through our GitHub Action:

- Mac: 3m 9s
- Windows: 8m 22s
- Ubuntu: 13m 22s

Ubuntu is spending 8 min just installing spirv-tools, this we should fix so we at least get down to 8m total build times instead of 13 min.

Just download the spriv-tools executables directly instead of using apt? think was some discussion in chat. cc @Jake-Shadle

**Comment by Jake-Shadle** [comment]
It was originally downloading the executables directly, but was changed in #63 to use brew instead, not sure why exactly, but you can see the CI runs before that PR were around ~6m which is still fairly high, but a lot better.

**Comment by repi** [comment]
Oh, does sound like we should go back!

**Comment by khyperia** [comment]
It looks like we're building at least some crates twice from the split `cargo test`. My guess:

current ci.yaml:

```yaml
      - run: cargo test --workspace --exclude example-runner
      - if: ${{ runner.os != 'macOS' }}
        run: cargo test --workspace example-runner
```

probably better ci.yaml:

```yaml
      - if: ${{ runner.os != 'macOS' }}
        run: cargo test --workspace
      - if: ${{ runner.os == 'macOS' }}
        run: cargo test --workspace --exclude example-runner
```

**Comment by khyperia** [comment]
now that CI is downloading spirv-tools instead of brew installing (and therefore building) them, is there anything left to do in this issue?

current times:

mac: 10m (2m installing spirv-tools, 7m running tests)
ubuntu: 9m30s (45 seconds installing native deps+spirv-tools, 7m30s running tests)
windows: 11m30s (13 seconds installing spirv-tools, 9m running tests)

seems like most time is spent in running tests (mostly building our boatload of dependencies, I think), so not really much we can improve on the CI side.

**Comment by Jake-Shadle** [comment]
We could add support for sccache, but that does add some complexity, I could do a quick and dirty pr to see if the speedup is worth it?

**Comment by khyperia** [comment]
I don't actually know, is sccache significantly better than the github cache action? We used to do that, but it was broken since rustup upgrades wouldn't invalidate the cache, so we removed it. Might be relatively easier to fix that instead. https://github.com/EmbarkStudios/rust-gpu/blob/8c9583f2c6e1fe69a34eac15c5591a1ef54dc166/.github/workflows/ci.yaml#L17-L26

(Or, y'know, the github cache action might have been doing absolutely nothing, because checked-out files are newer than the cache modification date, idk)
